Library/NodeClient.ts

```diff
@@ -0,0 +1,60 @@
+import http = require("http")
+import https = require("https")
+import Client = require("./Client")
+import ServerRequestTracking = require("../AutoCollection/ServerRequests")
+import ClientRequestTracking = require("../AutoCollection/ClientRequests")
+import NodeHttpDependencyTelemetry = require("./NodeHttpDependencyTelemetry")
+import NodeHttpRequestTelemetry = require("./NodeHttpRequestTelemetry")
+import Logging = require("./Logging")
+
+/**
+ * Application Insights telemetry client for Node.JS extends base Client object to provide Node.JS-specific methods
+ * to track incoming and outgoing HTTP requests.
+ */
+class NodeClient extends Client {
+
+    /**
+     * Log RequestTelemetry from HTTP request and response. This method will log immediately without waitng for request completion
+     * and it requires duration parameter to be specified on NodeHttpRequestTelemetry object.
+     * Use trackNodeHttpRequest function to log the telemetry after request completion
+     * @param telemetry Object encapsulating incoming request, response and duration information 
+     */
+    public trackNodeHttpRequestSync(telemetry: NodeHttpRequestTelemetry) {
+        if (telemetry && telemetry.request && telemetry.response && telemetry.duration) {
+            ServerRequestTracking.trackRequestSync(this, telemetry.request, telemetry.response, telemetry.duration, telemetry.properties, telemetry.error);
+        }
+        else {
+            Logging.warn("trackNodeHttpRequestSync requires NodeHttpRequestTelemetry object with request, response and duration specified.");
+        }
+    }
+
+    /**
+     * Log RequestTelemetry from HTTP request and response. This method will `follow` the request to completion.
+     * Use trackNodeHttpRequestSync function to log telemetry immediately without waiting for request completion
+     * @param telemetry Object encapsulating incoming request and response information
+     */
+    public trackNodeHttpRequest(telemetry: NodeHttpRequestTelemetry) {
```

Comment:
What's your opinion on simply overriding `trackRequest` here to overload? If input is not of type `NodeHttpRequestTelemetry` call into the base class. This allows `trackRequest` to be used for both manual and the preferred one of the two node shortcut methods. Same question for the dependency shortcut as well

Reply:
I am not sure I like the ambiguity it creates. If I know about AppInsights, track* is very simply to understand and I can see that extra methods are node-specific. If we use overloads, then I have no option but to learn about these additional overloads which adds complexity, so it is not that simple anymore.
Also I don't like the idea of checking the types for presence of specific fields, since these are interfaces so simple instanceof wouldn't work.

Reply:
Fair enough. I'm okay with having these separate methods.